### PR TITLE
Refactor and group createPullRequest input parameters

### PR DESCRIPTION
**Pull request by guillett**
Depends on #70

### CODE REVIEW

File: controllers/suggestionsController.js [hunk 1]
```diff
@@ -16,29 +16,28 @@ class SuggestionsController {
...
-	 @param title: (string) Will be used as a title for the pull request creation
-	 @param description: (urlencoded JSON string) The state we want to record
-	 @param state: (urlencoded JSON string) The state we want to record
+	 @param holder.title: (string) Will be used as a title for the pull request creation
+	 @param holder.description: The full commit message and PR initial comment
+	 @param holder.state: (urlencoded JSON string) The state we want to record
 	 @param res: An expressjs Response object to get back to the user
 	 */
-	createPullRequest(title, description, state, res) {
+	createPullRequest(holder, res) {
```
Comment (MattiSG) [inline controllers/suggestionsController.js]: This is not an acceptable name: anything can be a “holder”. You might as well call this “obj” or “lkjsdklsjglqskjg“ :wink:

What does this object model? What concept has a title, a description and a state? Is it a “testCase”? Is it a “testRun”? A “testSuggestion” maybe? 🙂

File: routers/main.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ router.post('/createSuggestion',
 	passport.authenticate(CREATE_PR_STRATEGY_NAME, {scope: [ 'repo' ]}));
 
 router.get('/github_callback/createPR', passport.authenticate(CREATE_PR_STRATEGY_NAME, {failureRedirect: '/authKO'}), (req, res) => {
-	suggestionsController.createPullRequest(req.session.title, req.session.description, req.session.state, res);
+	suggestionsController.createPullRequest(req.session, res);
```
Comment (MattiSG) [inline routers/main.js]: > The intention is not very clear here: the fact that the session is filled with the required properties is coincidental. I'd be in favour of whitelisting the wanted properties for the sake of intent readability. Here, I feel like you're going to manipulate the session itself.

File: test/controllers/suggestionsControllerSpec.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ describe('suggestionController', () => {
 				}
 			});
 			//action
-			const createPRPromise = suggestionsController.createPullRequest(title, description, state, res);
+			const createPRPromise = suggestionsController.createPullRequest(testData, res);
```
Comment (MattiSG) [inline test/controllers/suggestionsControllerSpec.js]: > “data” is very ambiguous. This could probably get a better name once `holder` above is renamed 🙂

**Comment by MattiSG** [comment]
> suggestionsCtrl.createPullRequest has 2 parameters now

This is not a good PR name. This describes an implementation and not an intent. Why are you switching to 2 parameters? Which problem is this solving? 🙂

**Comment by guillett** [comment]
I agree and I wanted to discuss the issue IRL but opportunities are limited.

First, those parameters should have be bundled together because they are properties of the same object.

The second point is about Ludwig's current architecture. Contributors should not be redirected to Ludwig once their contributions were taken into account but to the original application. (cf. WIP branch https://github.com/sgmap/ludwig/compare/refactor_create_pr...redirect).

File: test/controllers/suggestionsControllerSpec.js (hunk 3 above)
Comment (guillett) [inline test/controllers/suggestionsControllerSpec.js]: I recycled an already existing object.

**Pull request merged**